File: NDDD/NDDD.Domain/ValueObjects/AreaId.cs
using DDD.Domain.ValueObjects;

namespace NDDD.Domain.ValueObjects
{
    /// <summary>
    /// エリアID
    /// </summary>
    public sealed class AreaId : ValueObject<AreaId>
    {
        /// <summary>
        /// コンストラクタ
        /// </summary>
        /// <param name="value"></param>
        public AreaId(int value)
        {
            Value = value;
        }

        /// <summary>
        /// 値
        /// </summary>
        public int Value { get; }

        /// <summary>
        /// 表示する値
        /// </summary>
        public string DisplayValue => Value.ToString().PadLeft(4, '0');

        /// <summary>
        /// EqualsCore
        /// </summary>
        /// <param name="other">比較する値</param>
        /// <returns>同じときTrue</returns>
        protected override bool EqualsCore(AreaId other)
        {
            return this.Value == other.Value;
        }
    }
}

File: DDD 2/DDD.Domain/VB/ValueObjects/AreaId.vb
Imports System

Namespace DDD.Domain.ValueObjects
    Public NotInheritable Class AreaId
        Inherits ValueObject(Of AreaId)

        Public Sub New(ByVal value As Integer)
            Value = value
        End Sub

        Public ReadOnly Property Value As Integer

        Protected Overrides Function EqualsCore(ByVal other As AreaId) As Boolean
            Return Value = other.Value
        End Function

        Public ReadOnly Property DisplayValue As String
            Get
                Return Value.ToString().PadLeft(4, "0"c)
            End Get
        End Property
    End Class
End Namespace